enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,335 @@
+
+After the host reboots, the `kubelet` service will start the control plane static pods.
+Kubelet will send a CSR (see below) and join the cluster.
+CVO will deploy all cluster operators.
+The control plane operators will rollout a new revision (if necessary).
+
+#### OpenShift-installer
+
+We will add logic to the installer to create `bootstrap-in-place-for-live-iso.ign` Ignition config.
+This Ignition config will diverge from the default bootstrap Ignition:
+bootkube.sh:
+1. Start cluster-bootstrap without required pods (`--required-pods=''`)
+2. Run cluster-bootstrap with `--bootstrap-in-place` entrypoint to enrich the master Ignition.
+3. Write RHCOS image and the master Ignition to disk.
+4. Reboot the node.
+
+#### Cluster-bootstrap
+
+By default, `cluster-bootstrap` starts the bootstrap control plane and creates all the manifests under ``/opt/openshift/manifests``.
+`cluster-bootstrap` also waits for a list of required pods to be ready. These pods are expected to start running on the control plane nodes.
+In case we are running the bootstrap in place, there is no control plane node that can run those pods. `cluster-bootstrap` should apply the manifest and tear down the control plane. If `cluster-bootstrap` fails to apply some of the manifests, it should return an error.
+
+
+`cluster-bootstrap` will have a new entrypoint `--bootstrap-in-place` which will get the master Ignition as input and will enrich the master Ignition with control plane static pods manifests and all required resources including etcd data.
+
+#### Bootstrap / Control plane static pods
+
+The control plane components we will add to the master Ignition are (to be placed under `/etc/kubernetes/manifests`):
+
+1. etcd-pod
+2. kube-apiserver-pod
+3. kube-controller-manager-pod
+4. kube-scheduler-pod
+
+Control plane required resources to be added to the Ignition:
+
+1. `/var/lib/etcd`
+2. `/etc/kubernetes/bootstrap-configs`
+3. /opt/openshift/tls/* (`/etc/kubernetes/bootstrap-secrets`)
+4. /opt/openshift/auth/kubeconfig-loopback (`/etc/kubernetes/bootstrap-secrets/kubeconfig`)
+
+**Note**: `/etc/kubernetes/bootstrap-secrets` and `/etc/kubernetes/bootstrap-configs` will be deleted after the node reboots by the post-reboot service (see below), and the OCP control plane is ready.
+
+The control plane operators (that will run on the node post reboot) will manage the rollout of a new revision of the control plane pods.
+
+#### etcd data
+
+In order to add a viable, working etcd post reboot, we will take a snapshot of etcd and add it to the master Ignition.
+Post reboot we will use the restored etcd-member from the snapshot.
+
+Another option is to stop the etcd pod (move the static pod manifest from `/etc/kubernetes/manifests`).
+When stopped, etcd will save its state and exit. We can then add the `/var/lib/etcd` directory to the master Ignition config.
+After the reboot, etcd should start with all the data it had prior to the reboot.
```

Comment:
We need to add a hard requirement that the node IP is persisted through the reboot. If this is not true TLS SAN will be invalidated. This will cause communications between apiserver and etcd to fail.

Reply:
@hexfusion isn't it part of OCP requirements by default? Nothing should change.

Reply:
The change here is that it needs to get the same Ip it had while running from liveCD (after pivot it might have a new dhcp client ID) which isn't an OCP requirement.
 I'll document it.

Reply:
Added as requirement